src/test/java/pages/AmazonResultPage.java

```diff
@@ -0,0 +1,41 @@
+package pages;
+
+import org.openqa.selenium.By;
+import org.openqa.selenium.WebDriver;
+import org.openqa.selenium.WebElement;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Random;
+
+public class AmazonResultPage extends AmazonStartPage {
+
+    public AmazonResultPage (WebDriver driver) {
+        super(driver);
+    }
+
+    public int countResults() {
+        List<WebElement> results = fillResults();
+        return results.size();
+    }
+
+    public AmazonProductPage clickOnResult() {
```

Comment:
Bad naming. What result? Something like clickOnRandomItem would be more fitting.